### ISSUE TITLE
Autogenerated props tables are incorrect for numbers when using oneOf

### ISSUE THREAD

**Issue by StephenEsser** [cerner/terra-core#581]
# Issue Description

The props table generated for the read me and site are displaying incorrect results when using oneOf.

oneOf for numbers will list the data type as string
![screen shot 2017-07-06 at 2 44 34 pm](https://user-images.githubusercontent.com/6720991/27929793-f573c338-6259-11e7-80d1-6e76ceaf8d10.png)

## Issue Type

- [ ] New Feature
- [ ] Enhancement
- [x] Bug
- [ ] Other

## Expected Behavior

The documentation should display correct prop types for numbers when using oneOf

## Current Behavior

oneOf number will generate a prop table listing the data type as string

## Steps to Reproduce

1. Navigate to the terra site: http://engineering.cerner.com/terra-core/#/site/heading
2. View the table and see number props are listed as strings for oneOf
3. https://github.com/cerner/terra-core/blob/master/packages/terra-heading/src/Heading.jsx#L15

## Environment

* Component Version:
* Browser Name and Version:
* Operating System and version (desktop or mobile):

**Comment by Matt-Butler** [comment]
```
{
  "children": {
    "type": {
      "name": "node"
    },
    "required": true,
    "description": "Child nodes"
  },
  "level": {
    "type": {
      "name": "enum",
      "value": [
        {
          "value": "1",
          "computed": false
        },
        {
          "value": "2",
          "computed": false
        },
        {
          "value": "3",
          "computed": false
        },
        {
          "value": "4",
          "computed": false
        },
        {
          "value": "5",
          "computed": false
        },
        {
          "value": "6",
          "computed": false
        }
      ]
    },
    "required": true,
    "description": "Sets the heading level. One of `1`, `2`, `3`, `4`, `5`, `6`."
  },
  "color": {
    "type": {
      "name": "string"
    },
    "required": false,
    "description": "The color of the text. Accepts any color value parseable by CSS.",
    "defaultValue": {
      "value": "'inherit'",
      "computed": false
    }
  },
  "isItalic": {
    "type": {
      "name": "bool"
    },
    "required": false,
    "description": "Sets the text to display in italics.",
    "defaultValue": {
      "value": "false",
      "computed": false
    }
  },
  "isVisuallyHidden": {
    "type": {
      "name": "bool"
    },
    "required": false,
    "description": "Sets the text to appear visually hidden. The text will still be available to screen readers.",
    "defaultValue": {
      "value": "false",
      "computed": false
    }
  },
  "size": {
    "type": {
      "name": "enum",
      "value": [
        {
          "value": "'mini'",
          "computed": false
        },
        {
          "value": "'tiny'",
          "computed": false
        },
        {
          "value": "'small'",
          "computed": false
        },
        {
          "value": "'medium'",
          "computed": false
        },
        {
          "value": "'large'",
          "computed": false
        },
        {
          "value": "'huge'",
          "computed": false
        }
      ]
    },
    "required": false,
    "description": "Sets the text size. One of `mini`, `tiny`, `small`, `medium`, `large`, `huge`."
  },
  "weight": {
    "type": {
      "name": "enum",
      "value": [
        {
          "value": "200",
          "computed": false
        },
        {
          "value": "400",
          "computed": false
        },
        {
          "value": "700",
          "computed": false
        }
      ]
    },
    "required": false,
    "description": "Sets the text size. One of `200`, `400`, `700`.",
    "defaultValue": {
      "value": "700",
      "computed": false
    }
  }
}
```

react docgen will return props as a js object. We need to add logic to determine if a value is numeric.

**Comment by Matt-Butler** [comment]
We could replace this logic:
https://github.com/cerner/terra-core/blob/master/packages/terra-props-table/src/PropsTable.jsx#L50
with this:
```
          if (type === 'enum') {
            if (isNaN(prop.type.value[0].value)) {
              type = _typeof(prop.type.value[0].value);
            } else {
              type = 'number';
            }
          }
```

**Comment by mjhenkes** [comment]
resolved with #583
